call-home/src/bin/stats/main.rs

```diff
@@ -0,0 +1,165 @@
+};
+
+mod cache;
+mod collector;
+mod exporter;
+mod store;
+
+
+#[derive(Parser, Debug)]
+#[clap(name = utils::package_description!(), version = utils::version_info_str!())]
+struct Cli {
+    /// The url for mbus.
+    #[clap(long, short, default_value = "nats://mayastor-nats:4222")]
+    mbus_url: String,
+
+    /// Kubernetes namespace of mayastor service.
+    #[clap(global = true, long, short = 'n', default_value = "mayastor")]
+    namespace: String,
+}
+
+impl Cli {
+    fn args() -> Self {
+        Cli::parse()
+    }
+}
+
+/// Intilize mbus.
+pub(crate) async fn mbus_init(mbus_url: &str) -> errors::Result<NatsMessageBus> {
+    let message_bus: NatsMessageBus = message_bus_init(mbus_url).await;
+    Ok(message_bus)
+}
+
+/// Initialize events store cache from config map.
+pub(crate) async fn initialize_events_cache(init_data: ConfigMap) {
+    let events = EventSet::from_event_store(init_data).unwrap();
+    Cache::initialize(events);
+}
+
+/// Initialize events store.
+pub(crate) async fn initialize_events_store(namespace: &str) -> errors::Result<ConfigMap> {
+    let event_store: ConfigMap = initialize(namespace).await?;
+    Ok(event_store)
+}
+
+/// Initialize exporter config that are passed through arguments.
+fn initialize_exporter() {
+    ExporterConfig::initialize();
+}
+
+#[tokio::main]
+async fn main() -> errors::Result<()> {
+    init_logging();
+    info!("stats aggregation started!");
+
+    let args = Cli::args();
+
+    utils::print_package_info!();
+
+    let m_bus = mbus_init(args.mbus_url.as_str()).await?;
+    info!("mbus initialized successfully!");
+
+    let init_data = initialize_events_store(&args.namespace).await?;
+    info!("event store initialized successfully!");
+
+    initialize_events_cache(init_data).await;
+    info!("event cache initialized successfully!");
+
+    initialize_exporter();
+    info!("exporter initialized successfully!");
+
+    // spawn a new task to store the data in cache.
+    tokio::spawn(async move {
+        cache::events_cache::store_events(m_bus)
+            .await
+            .map_err(|error| {
+                error!(%error, "Error while storing the events to cahce");
+                flush_traces();
+                error
+            })
+    });
+
+    // spawn a new task to update the config map from cache.
+    tokio::spawn(async move {
+        store::events_store::update_config_map_data(&args.namespace)
+            .await
+            .map_err(|error| {
+                error!(%error, "Error while persisting the stats to config map from cache");
+                flush_traces();
+                error
+            })
+    });
+
+    let app = move || {
+        actix_web::App::new()
+            .wrap(middleware::Logger::default())
+            .configure(stats_route)
+    };
+
+    HttpServer::new(app)
+        .bind(exporter::exporter_config::ExporterConfig::get_config().metrics_endpoint())
+        .unwrap()
+        .run()
```

Comment:
Ah it may be that actix server already adds the shutdown by default, should be simple to test, check if goes down gracefully if you send sigterm.

Reply:
Since this code is working for exporter , can we have this code in ? Couldn't figure out how to send sigterm. 
Other PR on top of this will help in lesser code IMHO.